lib/onebox/engine/amazon_onebox.rb

```diff
@@ -45,7 +45,11 @@ def http_params
       private
 
       def match
-        @match ||= @url.match(/(?:d|g)p\/(?:product\/|video\/detail\/)?(?<id>[^\/]+)(?:\/|$)/mi)
+        @match ||= @url.match(/(?:d|g)p\/(?:product\/|video\/detail\/)?(?<id>[A-Z0-9]+)(?:\/|\?|$)/mi)
+      end
+
+      def link
```

Comment:
Is `#link` used anywhere? I see it overrides `Engine#link` (also a private method) but `Engine` doesn't use it on its own either.

Reply:
You're right - that shouldn't have been there.